### DISCUSSION TITLE
Disable unique or add sparse: true for fields within fields that have conditions

### DISCUSSION THREAD

**Discussion by ghardin1314** [payloadcms/payload#3785] [Feature Requests & Ideas]
### Link to reproduction

https://github.com/ghardin1314/payload/pull/1

### Describe the Bug

In previous versions of payload, when setting a group with a conditional statement, that conditional statement would also apply to all child fields. This means you could make a child `text` field with `unique` or `required` with no issues. This behavior seems to have changed in new versions of Payload

### To Reproduce

Create a new Post with `postType` of external. The first one should be created no problem. When creating a second Post with `postType` of external you should get the following error:

`The following field is invalid: internalProps.slug`

### Payload Version

v2.0.10

### Adapters and Plugins

db-mongo

**Comment by ghardin1314** [comment]
After further testing it seems like its just `unique` that causes an issue. `required` does seem to work as expected

**Comment by jmikrut** [comment]
Hey @ghardin1314 — good find. Do you need anything here still? Conditions do not, and likely should not, prevent unique from functioning accordingly but we may be able to set unique fields within conditions to `sparse: true`, which would then allow for multiple null / undefined values.

I don't think anything has changed here though. I may be wrong, and something did indeed change, but let me know what you think!

**Reply by ghardin1314** [reply]
Yes I think setting `sparse` to true would be the appropriate solution here. Otherwise it makes `unique` unusable for conditional fields since you will pretty much always have 2 items with `undefined` values. It appears this was the case in v1 and somehow changed in v2.

The alternative would be to expose `sparse` as an option that can be set by the user
